### PR TITLE
docs: Update docs for new connection methods

### LINKED ISSUE #250: [DOC] add docs for buildkit integration
### What kind of documentation improvement is needed?

Missing information or guidance

### What is the change that is needed?

add/update docs for #233 

https://github.com/project-copacetic/copacetic/tree/main/website/docs

### PR DESCRIPTION
**Pull request by cpuguy83**
Closes #250

### CODE REVIEW

File: website/docs/quick-start.md [hunk 1]
```diff
@@ -31,32 +31,37 @@ This sample illustrates how to patch containers using vulnerability reports with
...
+    You'll need a buildkit instance to connect to.
+    By default copa will attempt to auto-connect to an instance in order:
+        1. Default docker buildkit endpoint
+        2. Currently selected buildx builder (see: `docker buildx --help`)
+        3. buildkit daemon at the default address `/run/buildkit/buildkitd.sock`
+    
+    If an instance doesn't exist or that instance doesn't support all the features copa needs the next will be attempted.
+    You may need to specify a custom address using the `--addr` flag. Here are the supported formats:
+
+    - `unix:///path/to/buildkit.sock` - Connect to buildkit over unix socket.
```
Comment (sozercan) [inline website/docs/quick-start.md:45]: We should keep the instructions for this. Should we create a seperate section? wdyt?

File: website/docs/quick-start.md [hunk 2]
```diff
@@ -31,32 +31,37 @@ This sample illustrates how to patch containers using vulnerability reports with
...
+    - `tcp://$BUILDKIT_ADDR:$PORT` - Connec to buildkit over TCP. (not recommended for security reasons)
```
Comment (sozercan) [inline website/docs/quick-start.md:46]: We should keep the instructions for buildkit as a container or natively. Should we create a seperate section? wdyt?

File: website/docs/quick-start.md [hunk 3]
```diff
@@ -31,32 +31,37 @@ This sample illustrates how to patch containers using vulnerability reports with
     ```bash
     trivy image --vuln-type os --ignore-unfixed mcr.microsoft.com/oss/nginx/nginx:1.21.6
 
-3. Patch the image using the Trivy report. You will need to start `buildkitd` if it is not already running:
-
+3. Patch the image using the Trivy report.:
+
+    You'll need a buildkit instance to connect to.
```
Comment (sozercan) [inline website/docs/quick-start.md:36]:
```suggestion
3. Patch the image using the Trivy report and a buildkit instance to connect to:
```

File: website/docs/quick-start.md [hunk 4]
```diff
@@ -31,32 +31,37 @@ This sample illustrates how to patch containers using vulnerability reports with
...
-        "moby/buildkit:$BUILDKIT_VERSION" \
-        --addr tcp://0.0.0.0:$BUILDKIT_PORT
-    copa patch \
-        -i mcr.microsoft.com/oss/nginx/nginx:1.21.6 \
-        -r nginx.1.21.6.json \
-        -t 1.21.6-patched \
-        -a tcp://0.0.0.0:$BUILDKIT_PORT
+    Example: Connect to buildx
+    ```
+    docker buildx create --name demo
```
Comment (sozercan) [inline website/docs/quick-start.md:63]: do we need --use? Will this bootstrap automatically?

Comment (cpuguy83) [reply website/docs/quick-start.md:63]: Didn't want to do `--use` so as to show we are using the named instance.
The driver does bootstrap the builder automatically.

**Comment by sozercan** [comment]
https://github.com/project-copacetic/copacetic/pull/258/files#diff-9a47be8dffa25a00534765738842d84e008688252849af1c579c8b45b4943f7fR11 let's update this too

File: website/docs/quick-start.md [hunk 5]
```diff
@@ -31,32 +31,37 @@ This sample illustrates how to patch containers using vulnerability reports with
     ```bash
     trivy image --vuln-type os --ignore-unfixed mcr.microsoft.com/oss/nginx/nginx:1.21.6
 
-3. Patch the image using the Trivy report. You will need to start `buildkitd` if it is not already running:
-
+3. Patch the image using the Trivy report.:
+
+    You'll need a buildkit instance to connect to.
+    By default copa will attempt to auto-connect to an instance in order:
+        1. Default docker buildkit endpoint
```
Comment (sozercan) [inline website/docs/quick-start.md:38]: this requires 24.0 and the containerd-snapshotter, is that right? should we add this here or create a seperate section?

File: website/docs/quick-start.md (hunk 2 above)
Comment (cpuguy83) [reply website/docs/quick-start.md:45]: Added as examples

File: website/docs/quick-start.md (hunk 3 above)
Comment (cpuguy83) [reply website/docs/quick-start.md]: Updated

File: website/docs/quick-start.md (hunk 5 above)
Comment (cpuguy83) [reply website/docs/quick-start.md]: Added note here.

File: website/docs/quick-start.md [hunk 6]
```diff
@@ -31,32 +31,71 @@ This sample illustrates how to patch containers using vulnerability reports with
     ```bash
     trivy image --vuln-type os --ignore-unfixed mcr.microsoft.com/oss/nginx/nginx:1.21.6
 
-3. Patch the image using the Trivy report. You will need to start `buildkitd` if it is not already running:
-
+3. To patch the image, use the Trivy report and specify a buildkit instance to connect to:
+
+    By default copa will attempt to auto-connect to an instance in order:
+        1. Default docker buildkit endpoint (requires at least docker v24.0 with [containerd snapshotter]() support enabled)
```
Comment (sozercan) [inline website/docs/quick-start.md:37]: missing link?

**Review by sozercan — APPROVED** [review]
looks like missing a link for containerd snapshotter. lgtm otherwise, thank you!

File: website/docs/quick-start.md (hunk 6 above)
Comment (cpuguy83) [reply website/docs/quick-start.md]: 🤦 Fixed

**Pull request merged**